scripts/pylib/pytest-twister-harness/src/twister_harness/device/device_adapter.py

```diff
@@ -0,0 +1,263 @@
+        self._clear_internal_resources()
+
+        if not self.command:
+            self.generate_command()
+        self._flash_and_run()
+        self._device_run.set()
+        self._start_reader_thread()
+        self.connect()
+
+    def close(self) -> None:
+        """Disconnect, close device and close reader thread."""
+        if not self._device_run.is_set():
+            # device already closed
+            return
+        self.disconnect()
+        self._close_device()
+        self._device_run.clear()
+        self._join_reader_thread()
+
+    def connect(self) -> None:
+        """Start collecting output from device."""
+        if not self._is_device_running():
+            msg = 'Cannot connect to not working device'
+            logger.error(msg)
+            raise TwisterHarnessException(msg)
+        if self._is_device_connected():
+            logger.warning('Device already connected')
+            return
+        self._connect_device()
+        self._device_connected.set()
+
+    def disconnect(self) -> None:
+        """Stop collecting output from device."""
+        if not self._is_device_connected():
+            logger.warning("Device already disconnected")
+            return
+        self._disconnect_device()
+        self._device_connected.clear()
+
+    def readline(self, timeout: float | None = None, raise_timeout_exc: bool = True) -> str | object:
+        """Read line from device output"""
+        if timeout is None:
+            timeout = self.base_timeout
+        if self._is_device_connected() or not self._device_read_queue.empty():
+            data = self._read_from_queue(timeout, raise_timeout_exc)
+        else:
+            msg = 'No connection to the device and no more data to read.'
+            logger.error(msg)
+            raise TwisterHarnessException(msg)
+        return data
+
+    def readlines(self) -> list[str | object]:
+        """
+        Read all available output lines produced by device from internal buffer.
+        """
+        # give a moment for reader thread to collect output lines
+        time.sleep(0.1)
+        lines = []
+        while not self._device_read_queue.empty():
+            line = self.readline(0.1, raise_timeout_exc=False)
+            lines.append(line)
+        return lines
+
+    def flush(self) -> None:
+        """
+        Remove all available output produced by device from internal buffer.
+        """
+        self.readlines()
+
+    def write(self, data: bytes) -> None:
+        """Write data bytes to device."""
+        if not self._is_device_connected():
+            msg = 'No connection to the device'
+            logger.error(msg)
+            raise TwisterHarnessException(msg)
+        self._write_to_device(data)
+
+    def _initialize_log_files(self) -> None:
+        """
+        TODO:
+        Do we need this?
+        Maybe we should add to log file info about currently executed test?
+        What about multiple devices used in the same test?
```

Comment:
Jeśli planujemy dodawać kolejne dut-y (np. do testów z bsim), to może już teraz warto rozszerzyć nazwy plików o numer, np. handler_0.log, device_0.log. Albo pierwszy zostawić tak jak jest (czyli ten ładowany z fixtury, bez parametrów), a przy dodawaniu kolejnego dut-a, dodać numer.